### PR TITLE
BDN-473. AdUnits

### PR DESCRIPTION
**Pull request by klukwist**
https://appodeal.atlassian.net/browse/BDN-473
https://appodeal.atlassian.net/browse/BDN-474
https://appodeal.atlassian.net/browse/BDN-475

### CODE REVIEW

File: bidon/src/main/java/org/bidon/sdk/auction/usecases/impl/ExecuteRoundUseCaseImpl.kt [hunk 1]
```diff
@@ -179,18 +176,15 @@ internal class ExecuteRoundUseCaseImpl(
...
     ) {
         adSource.addRoundInfo(
             auctionId = auctionResponse.auctionId,
             roundId = round.id,
             demandAd = demandAd,
             roundIndex = roundIndex,
-            bidType = bidType
         )
         adSource.setStatisticAdType(adTypeParam.asStatisticAdType())
         adSource.addAuctionConfigurationId(
```
Comment (da2gl) [inline bidon/src/main/java/org/bidon/sdk/auction/usecases/impl/ExecuteRoundUseCaseImpl.kt]: can be rename to `addAuctionConfigurationUid`

File: bidon/src/main/java/org/bidon/sdk/adapter/AdAuctionParams.kt [hunk 2]
```diff
@@ -25,13 +28,14 @@ class AdAuctionParamSource(
...
     val timeout: Long,
-    private val lineItems: List<LineItem> = emptyList(),
-    private val onLineItemConsumed: (LineItem) -> Unit = {},
+    private val adUnits: List<AdUnit> = emptyList(),
+    private val onAdUnitsConsumed: (AdUnit) -> Unit = {},
 
     /**
      * Bid specific params
      */
     val json: JSONObject? = null,
```
Comment (da2gl) [inline bidon/src/main/java/org/bidon/sdk/adapter/AdAuctionParams.kt]: json is unnecessary?

File: bidon/src/main/java/org/bidon/sdk/segment/SegmentSynchronizer.kt [hunk 3]
```diff
@@ -10,18 +10,13 @@ internal interface SegmentSynchronizer {
     val attributes: SegmentAttributes
     val segmentUid: String?
 
-    @Deprecated("Use segmentUid instead")
-    val segmentId: String?
-
     /**
      * For parsing segmentId from responses /auction and /config-requests
```
Comment (da2gl) [inline bidon/src/main/java/org/bidon/sdk/segment/SegmentSynchronizer.kt]: comment can be change to `segmentUid`?

File: bidon/src/main/java/org/bidon/sdk/segment/SegmentSynchronizer.kt [hunk 4]
```diff
@@ -10,18 +10,13 @@ internal interface SegmentSynchronizer {
...
      */
-    fun parseSegmentId(rootJsonResponse: String)
+    fun parseSegmentUid(rootJsonResponse: String)
 
     /**
      * For reading previous segmentId from [KeyValueStorage]
```
Comment (da2gl) [inline bidon/src/main/java/org/bidon/sdk/segment/SegmentSynchronizer.kt]: comment can be change to `segmentUid`?

File: adapter/amazon/src/main/java/org/bidon/amazon/impl/BannerAuctionParams.kt [hunk 5]
```diff
@@ -3,13 +3,15 @@ package org.bidon.amazon.impl
...
 data class BannerAuctionParams(
     val bannerFormat: BannerFormat,
     val activity: Activity,
-    val slotUuid: String,
-    override val price: Double
+    override val adUnit: AdUnit,
+    override val price: Double,
 ) : AdAuctionParams {
-    override val lineItem: LineItem? = null
+    val slotUuid: String = requireNotNull(adUnit.extra?.getString("slotUuid")) {
```
Comment (da2gl) [inline adapter/amazon/src/main/java/org/bidon/amazon/impl/BannerAuctionParams.kt]: `slot_uuid`

File: adapter/amazon/src/main/java/org/bidon/amazon/impl/FullscreenAuctionParams.kt [hunk 6]
```diff
@@ -2,12 +2,14 @@ package org.bidon.amazon.impl
...
 
 data class FullscreenAuctionParams(
     val activity: Activity,
-    val slotUuid: String,
+    override val adUnit: AdUnit,
     override val price: Double
 ) : AdAuctionParams {
-    override val lineItem: LineItem? = null
-}
+    val slotUuid: String = requireNotNull(adUnit.extra?.getString("slotUuid")) {
```
Comment (da2gl) [inline adapter/amazon/src/main/java/org/bidon/amazon/impl/FullscreenAuctionParams.kt]: `slot_uuid`

File: adapter/bigoads/src/main/java/org/bidon/bigoads/impl/BigoBannerAuctionParams.kt [hunk 7]
```diff
@@ -3,24 +3,24 @@ package org.bidon.bigoads.impl
...
     val bannerFormat: BannerFormat,
-    val slotId: String,
     val bidPrice: Double,
-    val payload: String,
+    override val adUnit: AdUnit
 ) : AdAuctionParams {
     override val price: Double get() = bidPrice
-    override val lineItem: LineItem? = null
+    val payload: String = requireNotNull(adUnit.extra?.getString("payload"))
+    val slotId: String = requireNotNull(adUnit.extra?.getString("slotId"))
```
Comment (da2gl) [inline adapter/bigoads/src/main/java/org/bidon/bigoads/impl/BigoBannerAuctionParams.kt]: `slot_id`

File: adapter/bigoads/src/main/java/org/bidon/bigoads/impl/BigoBannerAuctionParams.kt [hunk 8]
```diff
@@ -3,24 +3,24 @@ package org.bidon.bigoads.impl
...
 data class BigoFullscreenAuctionParams(
-    val slotId: String,
     val bidPrice: Double,
-    val payload: String,
+    override val adUnit: AdUnit
 ) : AdAuctionParams {
     override val price: Double get() = bidPrice
-    override val lineItem: LineItem? = null
+    val payload: String = requireNotNull(adUnit.extra?.getString("payload"))
+    val slotId: String = requireNotNull(adUnit.extra?.getString("slotId"))
```
Comment (da2gl) [inline adapter/bigoads/src/main/java/org/bidon/bigoads/impl/BigoBannerAuctionParams.kt]: `slot_id`

File: adapter/mintegral/src/main/java/org/bidon/mintegral/MintegralAuctionParam.kt [hunk 9]
```diff
@@ -3,34 +3,38 @@ package org.bidon.mintegral
...
-    override val price: Double,
-    val payload: String,
-    val unitId: String?,
-    val placementId: String?,
+    bidResponse: BidResponse,
 ) : AdAuctionParams {
-    override val lineItem: LineItem? = null
+    override val adUnit: AdUnit = bidResponse.adUnit
+    override val price: Double = bidResponse.price
+    val payload: String = requireNotNull(bidResponse.extra?.getString("placement_id"))
```
Comment (da2gl) [inline adapter/mintegral/src/main/java/org/bidon/mintegral/MintegralAuctionParam.kt]: `payload` ?

File: adapter/mintegral/src/main/java/org/bidon/mintegral/MintegralAuctionParam.kt [hunk 10]
```diff
@@ -3,34 +3,38 @@ package org.bidon.mintegral
...
```
Comment (da2gl) [inline adapter/mintegral/src/main/java/org/bidon/mintegral/MintegralAuctionParam.kt]: `payload` ?

**Comment by da2gl** [comment]
also refactoring `ServerlessAuctionConfig`

File: bidon/src/main/java/org/bidon/sdk/adapter/AdAuctionParams.kt (hunk 2 above)
Comment (klukwist) [reply bidon/src/main/java/org/bidon/sdk/adapter/AdAuctionParams.kt]: not really. removed

**Review by AntonZarytski — APPROVED** [review]

**Pull request merged**